src/webapi/rendering_context.rs

```diff
@@ -27,7 +183,771 @@ impl RenderingContext for CanvasRenderingContext2d {
     }
 }
 
+impl CanvasGradient {
+
+    /// Adds a new stop, defined by an offset and a color, to the gradient. If the offset is 
+    /// not between 0 and 1, an INDEX_SIZE_ERR is raised, if the color can't be parsed as a 
+    /// CSS <color>, a SYNTAX_ERR is raised.
+    /// 
```

Comment:
There are no exceptions in Rust so talking about raising exceptions doesn't really have much point. (: